### PR TITLE
New: Add method decorators to AST (fixes #65)

### LINKED ISSUE #65: Add static and instance method decorators to the AST
As per https://github.com/eslint/typescript-eslint-parser/issues/63

### PR DESCRIPTION
**Pull request by JamesHenry**
Closes #65 

(Ignore the branch name! This PR relates to method decorators)

### CODE REVIEW

**Comment by JamesHenry** [comment]
All in good time, my dear @eslintbot

File: lib/ast-converter.js [hunk 1]
```diff
@@ -601,6 +601,29 @@ module.exports = function(ast, extra) {
...
+            return decorators.map(function(decorator) {
+                switch (decorator.expression.kind) {
+                    // Decorator Factory
+                    // case SyntaxKind.CallExpression: // fallthrough, no need to differentiate yet...
+                    // Decorator
+                    default:
+                        var exp = convertChild(decorator.expression);
+                        return {
+                            type: "Decorator",
+                            loc: exp.loc,
```
Comment (nzakas) [inline lib/ast-converter.js]: Based on Flow, we don't need this extra node.

File: lib/ast-converter.js [hunk 2]
```diff
@@ -995,6 +1018,10 @@ module.exports = function(ast, extra) {
                         static: Boolean(node.flags & ts.NodeFlags.Static),
                         kind: "method"
                     });
+
+                    if (node.decorators) {
```
Comment (nzakas) [inline lib/ast-converter.js]: In Flow, the decorators array is always present even if empty. Probably best to do that here.

**Comment by JamesHenry** [comment]
@nzakas Now ready for review, PTAL

**Comment by nzakas** [comment]
This looks great, nice work!

**Pull request merged**